src/http-spec.ts

```diff
@@ -7,8 +7,58 @@ import { IServer } from './servers';
  * HTTP Service
  */
 
+interface IOAuthFlow {
+  authorizationUrl: string;
+  tokenUrl: string;
+  refreshUrl?: string;
+  scopes: Dictionary<string>;
+}
+
+type SecurityDefinition =
+  | {
+      description?: string;
+    }
+  | {
+      type: 'apiKey';
+      name: string;
+      in: 'query' | 'header' | 'cookie';
+    }
+  | {
+      type: 'openIdConnect';
+      openIdConnectUrl: string;
+    }
+  | {
+      type: 'http';
+      scheme: string;
+      bearerFormat?: string;
+    }
+  | {
+      type: 'oauth2';
+      flows: {
+        implicit: IOAuthFlow;
+        password: IOAuthFlow;
+        clientCredentials: IOAuthFlow;
+        authorizationCode: IOAuthFlow;
+      };
+    };
+
 export interface IHttpService extends INode {
   name: string;
+  description?: string;
+  termsOfService?: string;
+  contact?: {
+    name?: string;
+    url?: string;
+    email?: string;
+  };
+  license?: {
+    name?: string;
+    url?: string;
+  };
+  version: string;
+  servers: Array<{ url: string; description?: string }>;
+  security?: Dictionary<string[], string>;
```

Comment:
Suggest required array for `security`, to match `IHttpOperation`. (can be empty array, this removes all sorts of undefined checks we'd otherwise have to sprinkle in our codebase)